### PR TITLE
Move postcopy migration logic to cmdlib

### PR DESCRIPTION
**Pull request by calumcalder**
Exposes hypervisor postcopy functions via RPC, and implements logic to execute them in the MigrateInstance tasklet

### CODE REVIEW

File: lib/cmdlib/instance_migration.py [hunk 1]
```diff
@@ -916,6 +917,10 @@ def _ExecMigration(self):
 
     self.feedback_fn("* starting memory transfer")
     last_feedback = time.time()
+
+    migration_caps = \
+      self.instance.hvparams.get(constants.HV_KVM_MIGRATION_CAPS, None)
```
Comment (Morgawr) [inline lib/cmdlib/instance_migration.py]: If you return [] instead of None in the get() function when we can't find the KVM migration caps, you don't need to test if the list is empty in the following postcopy_enabled = ... statement.

You can just do

migration_caps = self.instance.hvparams.get(constants.HV_KVM_MIGRATION_CAPS, [])
postcopy_enabled = 'postcopy-ram' in migration_caps

if it were None, it would except obviously so we prevent that here.

This has the advantage that postcopy_enabled will always be a boolean, otherwise in the situation of an empty list it would end up being [], which tests for false but is a different type and I don't like having the type of a variable change based on the context.

File: lib/server/noded.py [hunk 2]
```diff
@@ -710,6 +710,15 @@ def perspective_instance_migrate(params):
     return backend.MigrateInstance(cluster_name, instance, target, live)
 
   @staticmethod
+  def perspective_instance_start_postcopy(params):
+    """ Moves a migrating instance from precopy to postcopy mode
```
Comment (Morgawr) [inline lib/server/noded.py]: Nit: To keep a somewhat consistent docstring, use either "move" or "switch" both here and in the rpc_defs.py file.

File: lib/cmdlib/instance_migration.py (hunk 1 above)
Comment (calumcalder) [inline lib/cmdlib/instance_migration.py]: Good call. hvparams[HV_KVM_MIGRATION_CAPS] is a ':' delimited string, so I'll have it default to an empty string, which should serve the same purpose.
